testbeam_analysis/result_analysis.py

```diff
@@ -847,3 +848,55 @@ def calculate_efficiency(input_tracks_file, input_alignment_file, output_pdf, bi
                         out_pass[:] = total_track_density_with_DUT_hit.T
                         out_total[:] = total_track_density.T
     return efficiencies, pass_tracks, total_tracks
+
+
+def histogram_track_angle(input_tracks_file, output_track_angle_file, n_bins, use_duts=None, output_pdf=None):
+    '''Calculates and histograms the track angle of the fitted tracks for selected DUTs.
+    Parameters
+    ----------
+    input_tracks_file : string
+        file name with the tracks table
+    output_track_angle_file: string
+        output pytables file with fitted means and sigmas of track angles for selected DUTs
+    use_duts : iterable
+        The duts to plot/calculate the track angle. If None all duts in the input_tracks_file are used
+    n_bins: int
+        number of bins for track angle histogram
+    output_pdf : string
+        file name for the output pdf
+    '''
+    logging.info('=== Calculate track angles ===')
+
+    slopes = {}  # initialize dict for track slopes
+    with tb.open_file(input_tracks_file, 'r') as in_file_h5:
+        n_duts = len(in_file_h5.list_nodes("/"))  # determine number of DUTs
+
+        if use_duts is None:
+            use_duts = range(n_duts)
+        else:
+            use_duts = use_duts
+        dut_index = [('Tracks_DUT_%i' % i) for i in use_duts]
+
+        slope_0_temp = np.array([])
+        slope_1_temp = np.array([])
+        for node in in_file_h5.root:  # loop through all DUTs in track table
+            if node.name in dut_index:
+                for tracks_chunk, _ in analysis_utils.data_aligned_at_events(node, chunk_size=100000):  # only store track slopes of selected DUTs
```

Comment:
Yes much better. Chunk size should be a parameter of the function as it is for all other result analysis functions. A chunk_size of 499999 is usually used